pulsar/consumer_partition.go

```diff
@@ -570,15 +570,30 @@ func (pc *partitionConsumer) internalUnsubscribe(unsub *unsubscribeRequest) {
 
 func (pc *partitionConsumer) getLastMessageID() (*trackingMessageID, error) {
 	if state := pc.getConsumerState(); state == consumerClosed || state == consumerClosing {
-		pc.log.WithField("state", state).Error("Failed to redeliver closing or closed consumer")
-		return nil, errors.New("failed to redeliver closing or closed consumer")
+		pc.log.WithField("state", state).Error("Failed to getLastMessageID for the closing or closed consumer")
+		return nil, errors.New("failed to getLastMessageID for the closing or closed consumer")
 	}
-	req := &getLastMsgIDRequest{doneCh: make(chan struct{})}
-	pc.eventsCh <- req
+	backoff := &internal.DefaultBackoff{}
```

Comment:
I'm wondering why a `DefaultBackoff{}` works in tests.

But theoretically we should use `pc.options.backoffPolicy` and add a `IsMaxBackoffReached` method to the `BackoffPolicy` interface.

Reply:
I've explained it in the PR description. Adding a new method to the interface requires code changes for the user implementation BackoffPolicy. All users should add the implementation of `IsMaxBackoffReached` for their own.

We could redesign it in the next major version or `1.0.0`.

Reply:
But the `DefaultBackoff` object here might have a different policy with the user provided backoff.